.gitlab-ci.yml

```diff
@@ -475,206 +35,55 @@ trigger-contracts-ci:
-
-# have environment:url eventually point to the logs
+    - cd ./artifacts/$PRODUCT/
+    - VERSION="$(cat ./VERSION)"
+    - echo "${PRODUCT} version = ${VERSION}"
+    - test -z "${VERSION}" && exit 1
+    - cd ./chaostest
+    - npm link
+    - chaostest singlenodeheight -i $CONTAINER_IMAGE:$VERSION
 
-.deploy-cibuild:                   &deploy-cibuild
-  <<:                              *deploy
-  dependencies:
-    - publish-docker-substrate
 
-.deploy-tag:                       &deploy-tag
-  <<:                              *deploy
+.post-chaos-cleanup:
+  stage:                           post-chaos
   only:
-    variables:
-      - $DEPLOY_TAG
-
-# have environment:url eventually point to the logs
-
-deploy-ew3:
-  <<:                              *deploy-cibuild
-  environment:
-    name:                          parity-prod-ew3
-
-deploy-ue1:
-  <<:                              *deploy-cibuild
-  environment:
-    name:                          parity-prod-ue1
-
-deploy-ew3-tag:
-  <<:                              *deploy-tag
-  environment:
-    name:                          parity-prod-ew3
-
-deploy-ue1-tag:
-  <<:                              *deploy-tag
+    - harry/chaostest-init
+  tags: 
+    - kubernetes-parity-build
+  variables:
+    PRODUCT:                       substrate
+    DOCKERFILE:                    $PRODUCT.Dockerfile
+    CONTAINER_IMAGE:               parity/$PRODUCT
   environment:
-    name:                          parity-prod-ue1
-
-.validator-deploy:                 &validator-deploy
-  # script will fail if there is no artifacts/substrate/VERSION
-  <<:                              *build-only
-  stage:                           flaming-fir
-  dependencies:
-    - build-linux-substrate
-  image:                           parity/azure-ansible:v1
-  allow_failure:                   true
-  when:                            manual
-  tags:
-    - linux-docker
-
-validator 1 4:
-  <<:                              *validator-deploy
-  script:
-    - ./.maintain/flamingfir-deploy.sh flamingfir-validator1
-validator 2 4:
-  <<:                              *validator-deploy
-  script:
-    - ./.maintain/flamingfir-deploy.sh flamingfir-validator2
-validator 3 4:
-  <<:                              *validator-deploy
-  script:
-    - ./.maintain/flamingfir-deploy.sh flamingfir-validator3
-validator 4 4:
-  <<:                              *validator-deploy
+    name:                          parity-chaosnet
+  image:                           docker:stable
+  services:
+    - docker:dind
+  before_script:
+    - test "$Docker_Hub_User_Parity" -a "$Docker_Hub_Pass_Parity"
+        || ( echo "no docker credentials provided"; exit 1 )
+    - docker login -u "$Docker_Hub_User_Parity" -p "$Docker_Hub_Pass_Parity"
+    - docker info
   script:
-    - ./.maintain/flamingfir-deploy.sh flamingfir-validator4
+    - docker rmi $CONTAINER_IMAGE:$VERSION -f
```

Comment:
Why do the `docker login`. This only removes the image locally.

Reply:
Hi @mxinden, i finally hacked out a first successful CI flow attempt. Any new comments on the updated flow?